src/service/csmService.js

```diff
@@ -0,0 +1,198 @@
+import { GlobalConfig } from "../globalConfig";
+import {
+  getLdasEndpointUrl,
+  getCloudFrontSharedWorkerUrl,
+  getCsmJsUrl,
+  CHAT_WIDGET_METRIC_NAME_SPACE,
+  DEFAULT_WIDGET_TYPE
+} from "../configs/csmConfig";
+import { LogManager } from "../log";
+
+export const DIMENSION_CATEGORY = "Category";
+class CsmService {
+  constructor() {
+    this.widgetType = DEFAULT_WIDGET_TYPE;
+    this.logger = LogManager.getLogger({
+      prefix: "ChatJS-csmService"
+    });
+    this.csmInitialized = false;
+    this.metricsToBePublished = [];
+  }
+
+  async _loadCsmScript(url, headers) {
+    return await window.fetch(url, {
+      headers
+    }).then(response => response.text()).then((text) => {
+      return text;
+    })
+  };
+
+  async _loadCsmScriptAndExecute(stage, region, headers) {
+    try {
+      const url = getCsmJsUrl(stage, region);
+      const jsContent = await this._loadCsmScript(url, headers)
+      let script = document.createElement('script');
+      script.type = 'text/javascript';
+      script.innerHTML = jsContent
+      document.head.appendChild(script)
+    } catch (error) {
+      this.logger.error("load csm error: ", error);
+    }
+  };
+
+  initializeCSM() {
+    // avoid multiple initialization
+    if (this.csmInitialized) {
+      return;
+    }
+    const stage = GlobalConfig.getStage();
+    const region = GlobalConfig.getRegion();
+    const headers = {
+      'x-amzn-requested-by': 'chat-widget'
+    };
+
+    const sharedWorkerUrl = getCloudFrontSharedWorkerUrl(stage, region);
+    const ldasEndpoint = getLdasEndpointUrl(stage, region);
+    // try {
+      this._loadCsmScriptAndExecute(stage, region, headers).then(() => {
+        window.fetch(sharedWorkerUrl, { headers })
+          .then(response => response.text())
+          .then(text => {
+            const sharedWorkerBlobUrl = URL.createObjectURL(new Blob([text], { type: 'text/javascript' }));
+            let params = {
+              endpoint: ldasEndpoint,
+              namespace: CHAT_WIDGET_METRIC_NAME_SPACE,
+              sharedWorkerUrl: sharedWorkerBlobUrl,
+            };
+  
+            csm.initCSM(params);
+            this.logger.info(`CSMService is initialized in ${stage} ${region}`);
+            this.csmInitialized = true;
+            if (this.metricsToBePublished) {
+              this.metricsToBePublished.forEach((metric) => {
+                csm.API.addMetric(metric);
+              });
+              this.metricsToBePublished = null;
+            }
+          })
+          .catch(error => this.logger.error("CSM initialization failed:", error));
+      })
+    // } catch (error) {
```

Comment:
nit: commented lines

Reply:
removed commented line.
Updated screenshot in PR description.